addons/web/static/tests/views/list_view_tests.js

```diff
@@ -15682,4 +15682,18 @@ QUnit.module("Views", (hooks) => {
             assert.hasClass(target.querySelectorAll(".o_data_row")[1], "o_selected_row");
         }
     );
+
+    QUnit.test("check readonly class for readonly field", async function (assert) {
+        await makeView({
+            type: "list",
+            resModel: "foo",
+            serverData,
+            arch: `
+                <list>
+                    <field name="foo" readonly="1"/>
+                </list>
+            `,
+        });
+        assert.hasClass(target.querySelectorAll(".o_data_cell")[0], "o_readonly_modifier");
```

Comment:
Well, actually your assert could have simply been added in the "editable list view: readonly fields cannot be edited" test case.